src/zenml/integrations/kubernetes/orchestrators/kubernetes_orchestrator.py

```diff
@@ -88,6 +88,14 @@ def get_kube_client(
         if incluster is None:
             incluster = self.config.incluster
 
+        if incluster:
```

Comment:
If we use the incluster option, why is that better than using the service connector? Is there no expiration on the token that the incluster client uses?

Reply:
It's not "better" but it is the right thing to do.
Processes running in a pod should use the credentials that are made available to the pod through its service account. This ensures that only the permissions that the service account is allowed, area accessible to any functions that run inside the pod.

Currently, we use the credentials of an AWS IAM role that was used for the purpose of authenticating to the cluster. And that entity is mapped to the master role in our kubernetes cluster, which means it has all permissions in every namespace.

Reply:
I agree it's the right thing to do. Can it be confusing for users though because of the following scenario: They give ZenML certain permissions (using the service connector) but the service account used for the pod is missing some of those permissions which are needed to run the pipeline?

In any case, for the reason mentioned above this could be a breaking change for some users/scenarios right?

Reply:
The pod in question here is the orchestrator pod whose only purpose is to query the kubernetes api server to fetch information about the pod status of the different step pods. Thus, the permissions that it needs are kubernetes permissions to read pod resources. 

We create a custom k8s service account for this pod to use and give it the necessary permissions. Now, the code is implemented in a way that this service account is not being used at all. Had this been the intended behavior, we would not create this service account explicitly in the first place.

Reply:
The step code where the user might expect to use any permissions runs in different pods. These pods don't even have a custom k8s service account and just use the default. They are not intended to be used to access external resources (like AWS resources) directly. 

When a user wants to allow its pipelines to use external services, they should use service connectors, 
- either explicit methods, or 
- the implicit method where they leverage the IAM role we create for every tenant and can be used for their service connectors without providing any credentials.
- this IAM role currently has only permissions that we give but we can allow users to edit it.

Reply:
In other words, the EKS connector is only supposed to help users connect to the EKS resource and not all other AWS resources. This connector's only purpose is to allow the user to authenticate and submit k8s jobs to the cluster.

Any other IAM permissions are out of scope for this connector and it is not aware of them.

Reply:
This will also not lead to any breaking changes because the orchestrator pod's service account that we will now start using already has all the Kubernetes permissions that the orchestrator pod needs. And it also reflects on my tests so far.